lib/help_scout/api/access_token.rb

```diff
@@ -1,10 +1,40 @@
 # frozen_string_literal: true
 
+require 'date'
+
 module HelpScout
   class API
     class AccessToken
       class << self
```

Comment:
Opening this file looks like two full separate classes are being defined 🙃. What about pulling the singleton class, `class << self`, out into a separate class? It's currently standing at 48 lines and has no cohesion with the instance class; it's only called a single time from `.refresh!`. I think the only dependency is line 48 where it's calling `new` on the class… which might even read clearer with the full class name.

Overall, the singleton class has a really distinct purpose: retrieving an access token. It's essentially a factory object that abstracts away the various paths needed to get a token, including caching and HTTP. What about something like `AccessTokenRequest`?

Reply:
Addressed this in 7900d794e2b23744047f1918ce81234c4678b066 by extracting two more classes, `API::AccessToken::Cache` and `API::AccessToken::Request` to encapsulate some of the implementation details around each.